src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java

```diff
@@ -466,16 +447,11 @@ public void run() {
         }, SYNC_INTERVAL*1000, SYNC_INTERVAL*1000);
 
         // Add shutdown hook to flush cached prefs on normal termination
-        AccessController.doPrivileged(new PrivilegedAction<Void>() {
-            public Void run() {
-                Runtime.getRuntime().addShutdownHook(
-                    new Thread(null, null, "Sync Timer Thread", 0, false) {
-                    public void run() {
-                        syncTimer.cancel();
-                        syncWorld();
-                    }
-                });
-                return null;
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(null, null, "Sync Timer Thread", 0, false) {
```

Comment:
Future work, use a Virtual thread for this.

Reply:
https://bugs.openjdk.org/browse/JDK-8344793